### PR TITLE
Simplify: fix comparison with non-real argument

### PR DESCRIPTION
**Pull request by Costor**
#### Brief description of what is fixed or changed
If `Z` is non-real comparisons like `0 < Z` or `Z < 0` etc. fail with `TypeError: Invalid comparison of non-real Z` and should be rewritten as `Z.is_positive`  resp. `Z.is_negative` .

When testing qapply I hit an occurrence of this in simplify._reduce_inverses :
```
from sympy import simplify, symbols, Symbol, expand
x, y = symbols("x y", commutative=True)
Z = Symbol("Z", commutative=False) # or just real=False
p = simplify(2**Z * y**Z) # -> TypeError: Invalid comparison of non-real Z
print(p)
p = simplify(x**Z * y**Z) # -> TypeError: Invalid comparison of non-real Z
print(p)
```
and provided the fix. Also added a test case.

Question: Is there are particular reason why is the comparison `a < Z` not always translated to `(a - Z).is_negative` etc. which would remove these issues?
#### Release Notes

* simplify
  * fixed a comparison x < 0 to x.is_negative to avoid type error for non-real x

### CODE REVIEW

File: sympy/simplify/tests/test_simplify.py [hunk 1]
```diff
@@ -1046,3 +1046,12 @@ def test_issue_22210():
     d = Symbol('d', integer=True)
     expr = 2*Derivative(sin(x), (x, d))
     assert expr.simplify() == expr
+
```
Comment (sylee957) [inline sympy/simplify/tests/test_simplify.py:1049]:
```suggestion


```

**Comment by oscarbenjamin** [comment]
> Question: Is there are particular reason why is the comparison `a < Z` not always translated to `(a - Z).is_negative` etc. which would remove these issues?

The behaviour of `x < y` for sympy expressions is problematic. There are two basic problems:

1. Writing `if x < y` gives a TypeError if the condition is not symbolically resolvable.
2. If either side is non-real or nan then we get a TypeError.

I think that point 1 is unavoidable but at least we can workaround with
```
if (x < y) is S.true:
```
The problem with point 2 is more pernicious because only catching the exception can help. There are issues around that discuss this but I think that ideally `x < y` would just be False if either x or y is non-real.

Writing `(x - y).is_negative` in most places probably would fix a bunch of bugs. Note that there is also a dedicated function `is_lt(x, y)` for this. With `is_lt` no exception is raised:
```python
>>> print(is_lt(I, S.One))
None
>>> print(is_lt(nan, S.One))
None
>>> print(is_lt(Integer(2), S.One))
False
```

One other point which might be worth considering in this PR is that `(x - y).is_negative = True` is not exactly equivalent to `(x < y) = True`. The reason is that `negative` is considered to refer only to finite numbers. You need `is_extended_negative`:
```python
>>> (1 - oo).is_negative
False
>>> (1 - oo).is_extended_negative
True
```
The `is_lt` function checks for extended negative.

**Comment by Costor** [comment]
Sorry I hadn't noticed this PR hadn't been closed. I've changed `args[1].is_negative` to `args[1].is_extended_negative` which I find easier to read than `is_lt(args[1], S.Zero)`  ( `is_lt(args[1], 0)` gives a type error).

**Comment by oscarbenjamin** [comment]
Looks good.

**Pull request merged**